rpc-rs/src/provider.rs

```diff
@@ -335,109 +337,124 @@ impl HostBridge {
-                                                &inv.operation, &inv.origin.public_key, e,
+        tokio::spawn(
+            async move {
+                while let Some(msg) = sub.next().await {
+                    match deserialize::<Invocation>(&msg.data) {
+                        Ok(mut inv) => {
+                            match this.dechunk_validate(&mut inv).await {
+                                Ok(()) => {
+                                    let provider = provider.clone();
+                                    let rpc_client = this.rpc_client().clone();
+                                    tokio::task::spawn(
+                                        async move {
+                                            tracing::span::Span::current().record(
+                                                "operation",
+                                                &tracing::field::display(&inv.operation),
                                             );
-                                            InvocationResponse {
-                                                invocation_id: inv.id,
-                                                error: Some(e.to_string()),
-                                                ..Default::default()
+                                            tracing::span::Span::current().record(
+                                                "public_key",
+                                                &tracing::field::display(&inv.origin.public_key),
+                                            );
+                                            trace!("Dispatching RPC invocation");
+                                            let response = match provider
+                                                .dispatch(
+                                                    &Context {
+                                                        actor: Some(inv.origin.public_key.clone()),
+                                                        ..Default::default()
+                                                    },
+                                                    Message {
+                                                        method: &inv.operation,
+                                                        arg: Cow::from(inv.msg),
+                                                    },
+                                                )
+                                                .await
+                                            {
+                                                Ok(msg) => InvocationResponse {
+                                                    invocation_id: inv.id,
+                                                    msg: msg.arg.to_vec(),
+                                                    ..Default::default()
+                                                },
+                                                Err(e) => {
+                                                    error!(
```

Comment:
there are fewer fields logged here than there used to be - are they added automatically by the trace span above?

Reply:
Yep they are!